src/shout_subgroup/group_chat_listener.py

```diff
@@ -88,41 +88,46 @@ async def listen_for_messages_handler(update: Update, context: ContextTypes.DEFA
         last_name=update.message.from_user.last_name
     )
 
-    session = get_database()
-    try:
-        maybe_added_user = await add_user_to_group_chat(session, update.effective_chat, user_who_sent_the_message)
-        # TODO: Add message like "The bot has recognized John Doe"
-        return
-    except NotGroupChatError:
-        # await update.message.reply_text("Sorry, you can only create or modify subgroups in group chats.")
-        return
+    db_session = get_database()
+
+    with db_session.begin() as session:
```

Comment:
We get the session factory from `get_database()` then use `.begin()` to start the session. When we use `.begin()` within the context of a `with` block. We get the `rollback()` and `commit()` handling for free.

https://docs.sqlalchemy.org/en/20/orm/session_basics.html#using-a-sessionmaker